biowdl.wdl

```diff
@@ -73,3 +73,48 @@ task InputConverter {
         json: {description: "JSON file version of the input sample sheet."}
     }
 }
+
+task IndexFastaFile {
+    input {
+        File inputFile
+        String javaXmx = "2G"
+        String memory = "3GiB"
+    }
+    String outputFile = basename(inputFile)
+    # Capture .fa¸ .fna and .fasta
+    String outputDict = sub(outputFile, "\.fn?as?t?a?$", "") + ".dict"
+    # This executes both picard and samtools, so indexes are co-located in the same folder.
+    command <<<
+        set -e
+        cp ~{inputFile} ~{outputFile}
+        picard -Xmx~{javaXmx} \
+            -XX:ParallelGCThreads=1 \
+            CreateSequenceDictionary \
+            REFERENCE=~{inputFile} \
+            OUTPUT="~{outputDict}"
+        samtools faidx ~{outputFile} --fai-idx ~{outputFile}.fai
+    >>>
+
+    output {
+        File outputFasta = outputFile
+        File outputFastaDict = outputDict
+        File outputFastaFai = outputFile + ".fai"
+    }
+
+    runtime {
+        memory: memory
+        # Contains picard 2.27.4, samtools 1.15.1
+        docker: "quay.io/biocontainers/mulled-v2-b0664646864bfdb46c5343b1b2b93fc05adb4b77:39a005770a3e30fb6aa3bf424b57ddf52bae7ece-0"
```

Comment:
samtools also has a command to produce a dict file. Using that would allow us to use a more straight forward (and probably smaller) container than this mulled one.

Reply:
Great find! I wish I had known that before.
https://www.htslib.org/doc/samtools-dict.html